### DISCUSSION TITLE
Why Was Quasar Chosen for Genie?

### DISCUSSION THREAD

**Discussion by TheCedarPrince** [GenieFramework/Stipple.jl#92] [Q&A] [answered]
Out of curiosity, up until working with the GenieFramework, I had no idea about Quasar.
Why was Quasar chosen?
It seems a pretty robust library but I just worry about its continuity and if/when it may be deprecated or go out of date.
Thanks!

**Comment by essenciary** [comment, marked as answer]
Mostly for the great docs and some advanced features in some complex components (like the data table). 

I wouldn't worry about continuity. v2 was just released with Vue 3 support and it's actively developed (practically daily commits).

Also have you checked their Github stats? 
* 510 contributors
* used by some 20K projects

If everything else fails, the Stipple ecosystem is open so anybody can add support for extra UI packages, either specialized ones (ex https://www.ag-grid.com/vue-data-grid/) or full UI libraries (https://vuetifyjs.com/en/).

**Reply by hhaensel** [reply]
My two cents: Already now you can easily add any component by the following routines:
```julia
julia> vue(:elem, "test", :me, param="eternal", [" more text"])
"<vue-elem param=\"eternal\" me>test more text</vue-elem>"

julia> xelem(:elem, "test", :me, param="eternal", [" more text"])
"<elem param=\"eternal\" me>test more text</elem>"

julia> xelem(:my__elem, "test", :me, param="eternal", [" more text"])
"<my-elem param=\"eternal\" me>test more text</my-elem>"
```
EDIT: dashs can be encoded as double underscore

**Reply by TheCedarPrince** [reply]
This was super helpful to learn about @essenciary !
I was just wondering what your perspective on it was here and I am excited to see how Genie continues to support Quasar!

Also, @hhaensel , what does that do?
I am having a bit of difficulty seeing how that could go into a webapp in Genie - are there docs anywhere on this to show an example of what you are doing here?
Thanks!

**Reply by hhaensel** [reply]
The `Stipple.jl` package is designed to build web applications based on vue-js. Vue-js is one of the big javascript frameworks like React or Angular and there is very low probability that it will stop being continued.

Quasar is a higher-level API that is built on top of vue-js and StippleUI provides an interface to many of the quasar elements. However, any other vue-js element can be used in the web-app. Therefore, I built the three functions above that make it easy to integrate such elements into your user interface. So, if e.g., the tooltip element is not (yet) defined by Stipple, you can still look up its definition on the [quasar page](https://quasar.dev/vue-components) and then build your own tooltip function.
```julia
tooltip(args...; kwargs...) = quasar(:tooltip, args...; kwargs...)
```
But if ever Quasar stops being supported we can still switch to a different Vue-js based UI like Vuetify without much hassle.
